engines/stark/services/userinterface.cpp

```diff
@@ -396,4 +399,31 @@ void UserInterface::performToggleSubtitle() {
 	_toggleSubtitle = false;
 }
 
+void UserInterface::cycleInventory(int step) {
+	Resources::KnowledgeSet *inventory = StarkGlobal->getInventory();
+	Common::Array<Resources::Item *> inventoryItems = inventory->listChildren<Resources::Item>(Resources::Item::kItemInventory);
+
+	int16 curItem = getSelectedInventoryItem();
+	int16 nextItem = curItem + step;
```

Comment:
The original game has a very strange feature: when you start to cycle the inventory with nothing selected at the beginning, it will skip the first (or the last) item. This happens only in the start and the skipped item will show when you keep on pressing A or S. I can't think of a reasonable way to implement this and it feels like a "bug" to me, so I didn't do it here.